src/main/java/com/thealgorithms/others/StringMatchFiniteAutomata.java

```diff
@@ -1,80 +1,126 @@
 package com.thealgorithms.others;
 
+import java.util.Set;
+import java.util.TreeSet;
+
 /**
- * @author Prateek Kumar Oraon (https://github.com/prateekKrOraon)
+ * A class to perform string matching using <a href="https://en.wikipedia.org/wiki/Finite-state_machine">finite automata</a>.
+ *
+ * @author <a href="https://github.com/prateekKrOraon">Prateek Kumar Oraon</a>
  */
-import java.util.Scanner;
-
-// An implementation of string matching using finite automata
 public final class StringMatchFiniteAutomata {
-    private StringMatchFiniteAutomata() {
-    }
-
-    public static final int CHARS = 256;
-    public static int[][] fa;
-    public static Scanner scanner = null;
-
-    public static void main(String[] args) {
-        scanner = new Scanner(System.in);
-        System.out.println("Enter String");
-        String text = scanner.nextLine();
-        System.out.println("Enter pattern");
-        String pat = scanner.nextLine();
 
-        searchPat(text, pat);
+    // Constants
+    private static final int CHARS = 256; // Total number of characters in the input alphabet
```

Comment:
```suggestion
    private static final int CHARS = Character.MAX_VALUE + 1; // Total number of characters in the input alphabet
```
`char` has 16 bits in java. Therefore maybe one could consider using a `Map`-like structure to represent the state transition table.